Cython/Compiler/ModuleNode.py

```diff
@@ -282,31 +283,35 @@ def h_entries(entries, api=0, pxd=0):
             h_guard = self.api_name(Naming.h_guard_prefix, env)
             h_code_start.put_h_guard(h_guard)
             h_code_start.putln("")
+            self._put_setup_code(h_code_start, "InitLimitedAPI")
             h_code_start.putln('#include "Python.h"')
-            self.generate_type_header_code(h_types, h_code_start)
+            self.generate_type_header_code(h_types, h_code_main)
```

Comment:
I'm a little wary here of how much `h_code_start`->`h_code_main` might change the order... we'll see